### DISCUSSION TITLE
Feature request: Organize Folders and pictures (Write support)

### DISCUSSION THREAD

**Discussion by kruemelro** [photoview/photoview#919] [General]
It would be very comfortable to be able to create albums and organize pictures inside photoview. Cause the filesystem is the only source of truth this means photoview have to be able to write to the filesystem.

Summary of our discord discussion:

- write support should be optional cause of security issues, maybe there should be a hint to the admin to make him aware of the risk
- Linux filesystem movements should be safe enough but it should be double checked to really get a rock solid solution. Its not acceptable to loose a single picture
- It should be checked how to move multiple pictures at once safely
- There also should be an option for each user if he wants to get asked before anything will me moved
- drag and drop support in the frontend would be the cherry on the cake :-)

**Comment by viktorstrate** [comment]
An alternative to writing to disk is organisation through virtual albums and collections, as proposed in #414.

**Comment by aparatext** [comment]
I think deleting images from Photoview is quite important as well.

**Comment by rexzhang** [comment]
> I think deleting images from Photoview is quite important as well.

maybe, we can move the photo file to a special folder instead delete the operation. this special folder will be not displayed in the Photoview.

Photoview doesn't remove any file, just only move ;-)

**Comment by wlanut** [comment]
Has there been any updates on this feature request? I do like the ability to maintain a file structure, but at the same time, I don't want less technically savvy users to have to use a separate interface for managing photos.

**Comment by kkovaletp** [comment]
As the progress of implementing a file management functionality is absent for several years here, I use a separate self-hosted web file manager for that. This is a pretty obvious solution, but nobody mentioned it yet...

**Comment by degudejung** [comment]
Sorry for jumping on this old thread. I tested photoview and truly love its simplicity and performance (even with just a simple sqlite). I also appreciate the single-source-of-truth principle regarding ro vs writing to the fs.

In the end, I opted for PhotoPrism even though it is much heavier and is not my first choice from a privacy perspective. I need to be honest though; if I want my family to dump photos of value from their iPhones or Macs to the family home server, to become somewhat independent from iCloud, then that requires a simple/comfortable solution. Not being able to sync is totally ok but using a third app for the upload - they just won't do it as much I want the private archive to grow. Maybe all that is not a new thought I still wanted to share are simple user's decision process anyways. Still, great tool, best of luck!
